src/app/contexts/Migration/Migration.tsx

```diff
@@ -88,6 +89,37 @@ export const MigrationProvider = ({ children }: Properties) => {
 	const [contract, setContract] = useState<Contract>();
 	const [contractIsPaused, setContractIsPaused] = useState<boolean>();
 
+	const getContractMigrations = useCallback(
+		async (transactionIds: string[]) => {
+			let contractMigrations: ARKMigrationViewStructOutput[] = [];
+
+			try {
+				contractMigrations = await contract!.getMigrationsByArkTxHash(transactionIds);
+			} catch {
+				//
+			}
+
+			return contractMigrations;
+		},
+		[contract],
+	);
+
+	const getTransactionStatus = useCallback(
+		async (transaction: DTO.ExtendedConfirmedTransactionData) => {
+			const contractMigrations = await getContractMigrations([`0x${transaction.id()}`]);
+
+			const contractMigration = contractMigrations.find(
+				(contractMigration: ARKMigrationViewStructOutput) =>
+					contractMigration.arkTxHash === `0x${transaction.id()}`,
+			);
+
+			return contractMigration!.recipient === ethers.constants.AddressZero
```

Comment:
@alfonsobries what will happen if the contractMigration is undefined here? It might stuck on loading state? :hmhm:

Reply:
can be part of https://app.clickup.com/t/86774j27t instead of this PR, we may need handling in more places